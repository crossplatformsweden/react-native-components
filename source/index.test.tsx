import React from 'react';
import TestRenderer from 'react-test-renderer';
import { CrossButton } from './index';

function setup(
  iconName: string,
  title?: any,
  size?: number,
  onPress?: () => void
) {
  return TestRenderer.create(
    <CrossButton
      iconName={iconName}
      title={title}
      color='blue'
      onPress={onPress || jest.fn()}
      size={size || 64}
    />
  );
}

jest.unmock('react-native');

describe('components', () => {
  describe('Index should have <CrossButton />', () => {
    it('Component should render', () => {
      const wrapper = setup('home', 'home');
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
