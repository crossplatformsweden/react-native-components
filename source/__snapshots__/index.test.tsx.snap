// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Index should have <CrossButton /> Component should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Button
    color="blue"
    disabled={false}
    icon={[Function]}
    iconName="home"
    mode="text"
    onPress={[MockFunction]}
    size={64}
    style={
      Object {
        "height": 45,
        "marginRight": 10,
      }
    }
    title="home"
  >
    home
  </Button>
</View>
`;
